### DISCUSSION TITLE
Support for conditional if in const

### DISCUSSION THREAD

**Discussion by serkonda7** [vlang/v#11717] [Ideas]
I imagine something like this
```v
const (
	foo = $if linux {
		'lnx'
	} $else $if windows {
		'win'
	} $else $if macos {
		'mac'
	} $else {
		'other'
	}
)
```

It could be useful if you want to have some platform-specific constant settings, like a filepath.

**Comment by dumblob** [comment]
Hm, IIRC there were some attempts in *V* to itroduce compile-time evaluation. I don't see it documented though, so it might be only a stub for internal use. Maybe @medvednikov and other core devs could comment more on this.

**Comment by spaceface777** [comment]
This will work in the future, the only reason it doesn't work right now is because `$if` expressions haven't been implemented as `const` initializations yet.
